samtranslator/swagger/swagger.py

```diff
@@ -93,15 +93,17 @@ def add_disable_execute_api_endpoint_extension(self, disable_execute_api_endpoin
         else:
             SwaggerEditor._update_dict(self._doc, self._X_ENDPOINT_CONFIG, set_disable_api_endpoint)
 
-    def add_lambda_integration(  # type: ignore[no-untyped-def]
```

Comment:
note: I moved the validation out so that SwaggerEditor doesn't need to know about logical ID